File: Python solutions/786. K-th Smallest Prime Fraction.py
class Solution:
    def kthSmallestPrimeFraction(self, arr: List[int], k: int) -> List[int]:
        pairs = []
        n = len(arr)
        for i in range(n):
            for j in range(i + 1, n):
                fraction = arr[i] / arr[j]
                pair = (fraction, (arr[i], arr[j]))
                pairs.append(pair)
        
        pairs = sorted(pairs, key=lambda x: x[0])
        ans = []
        ans.append(pairs[k - 1][1][0])
        ans.append(pairs[k - 1][1][1] )

        return ans
            
        

        

File: C++ solutions/786. K-th Smallest Prime Fraction.cpp
class Solution {
public:
    vector<int> kthSmallestPrimeFraction(vector<int>& arr, int k) {
        vector<pair<double, pair<int, int>>> pairs;
        int n = arr.size();
        
        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                double fraction = static_cast<double>(arr[i]) / arr[j];
                pairs.push_back({fraction, {arr[i], arr[j]}});
            }
        }
        
        sort(pairs.begin(), pairs.end());
        
        vector<int> ans;
        ans.push_back(pairs[k - 1].second.first);
        ans.push_back(pairs[k - 1].second.second);
        
        return ans;
    }
};